sender.go

```diff
@@ -64,29 +65,60 @@ type (
 		tab.Carrier
 		toMsg() (*amqp.Message, error)
 	}
-)
 
-// newSender creates a new Service Bus message sender given an AMQP client and entity path
-func (h *Hub) newSender(ctx context.Context) (*sender, error) {
-	span, ctx := h.startSpanFromContext(ctx, "eh.sender.newSender")
-	defer span.End()
+	// amqpSender is the bare minimum we need from an AMQP based sender.
+	// (used for testing)
+	amqpSender interface {
+		Send(ctx context.Context, msg *amqp.Message) error
+		Close(ctx context.Context) error
+	}
 
-	s := &sender{
-		hub:         h,
-		partitionID: h.senderPartitionID,
+	// getAmqpSender should return a live sender (exactly mimics the `amqpSender()` function below)
+	// (used for testing)
+	getAmqpSender func() amqpSender
+
+	senderRetryOptions struct {
+		recoveryBackoff *backoff.Backoff
+
+		// maxRetries controls how many times we try (in addition to the first attempt)
+		// 0 indicates no retries, and < 0 will cause infinite retries.
+		// Defaults to -1.
+		maxRetries int
+	}
+)
+
+func newSenderRetryOptions() *senderRetryOptions {
+	return &senderRetryOptions{
 		recoveryBackoff: &backoff.Backoff{
 			Min:    10 * time.Millisecond,
 			Max:    4 * time.Second,
 			Jitter: true,
 		},
-		cond: sync.NewCond(&sync.Mutex{}),
+		maxRetries: -1, // default to infinite retries
+	}
+}
+
+// newSender creates a new Service Bus message sender given an AMQP client and entity path
+func (h *Hub) newSender(ctx context.Context, retryOptions *senderRetryOptions) (*sender, error) {
+	span, ctx := h.startSpanFromContext(ctx, "eh.sender.newSender")
+	defer span.End()
+
+	if retryOptions == nil {
```

Comment:
Minor issue, and I know it sounds weird, but the idiom in Go is to _not_ check this for `nil` and let it panic.

Reply:
Done. I'm here for the idioms :)